lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1918,310 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
+            resource_group_name,
+            vm_name,
+            params_create,
+        )
+        vm_result = vm_poller.result()
+        object_id_vm = vm_result.identity.principal_id
+        log.info(object_id_vm)
+
+        if not object_id_vm:
+            raise ValueError("object_id_vm is not set.")
+
+        return object_id_vm
+
+    except Exception as e:
+        log.error(f"Error creating VM: {e}")
+        raise LisaException(str(e))
+
+
+def get_key_vault_management_client(
+    credential: DefaultAzureCredential, subscription_id: str
+) -> KeyVaultManagementClient:
+    return KeyVaultManagementClient(credential, subscription_id)
+
+
+def create_keyvault(
+    credential: DefaultAzureCredential,
+    subscription_id: str,
+    user_tenant_id: str,
+    user_object_id: str,
+    object_id_vm: str,
+    location: str,
+    resource_group_name: str,
+    vault_name_a: str,
+) -> Any:
+    keyvault_client = get_key_vault_management_client(credential, subscription_id)
+
+    vault_properties = VaultProperties(
+        tenant_id=user_tenant_id,
+        sku=KeyVaultSku(name="standard"),
+        access_policies=[
+            AccessPolicyEntry(
+                tenant_id=user_tenant_id,
+                object_id=user_object_id,
+                permissions=Permissions(
+                    keys=["all"], secrets=["all"], certificates=["all"]
+                ),
+            ),
+            AccessPolicyEntry(
+                tenant_id=user_tenant_id,
+                object_id=object_id_vm,
+                permissions=Permissions(
+                    keys=["all"], secrets=["all"], certificates=["all"]
+                ),
+            ),
+        ],
+    )
+
+    parameters = VaultCreateOrUpdateParameters(
+        location=location, properties=vault_properties
+    )
+    keyvault_poller = keyvault_client.vaults.begin_create_or_update(
+        resource_group_name, vault_name_a, parameters
+    )
+
+    return keyvault_poller.result()
+
+
+def delete_keyvault(
+    credential: Any,
+    subscription_id: str,
+    resource_group_name: str,
+    vault_name_a: str,
+    log: Logger,
+) -> None:
+    keyvault_client = get_key_vault_management_client(credential, subscription_id)
+
+    try:
+        keyvault_poller = keyvault_client.vaults.delete(
+            resource_group_name, vault_name_a
+        )
+        log.info(f"{keyvault_poller}")
+        log.info(f"Key Vault {vault_name_a} deleted successfully.")
+    except Exception as e:
+        log.error(f"Error deleting Key Vault: {e}")
+        raise LisaException(str(e))
+
+
+def get_key_vault_access_control_client(
+    vault_url: str, credential: DefaultAzureCredential
+) -> KeyVaultAccessControlClient:
+    return KeyVaultAccessControlClient(vault_url=vault_url, credential=credential)
+
+
+# If RBAC policy is needed/wanted for Key Vault
+# Remember then to set up enable_rbac_role = True
+# In Vault properties
+def assign_role_to_principal(
```

Comment:
seems it is not used

Reply:
it adds flexibility for users to decide in which type of access policies they want to implement on their key vault, i can delete but chance of someone adding this in the future very likely